src/overlays/actors/ovl_Boss_Va/z_boss_va.c

```diff
@@ -173,157 +319,3695 @@ static ColliderQuadInit D_8095BFF4 = {
+    if (sDoorState == 29) {
+        Audio_PlayActorSound2(&this->actor, NA_SE_EV_BUYODOOR_CLOSE);
+    }
+
+    if (sCsState <= INTRO_DOOR_SHUT) {
+        if (sDoorState < 100) {
+            sDoorState += 8;
+        } else {
+            sDoorState = 100;
+        }
+    }
+}
+
+void BossVa_Update(Actor* thisx, GlobalContext* globalCtx2) {
+    GlobalContext* globalCtx = globalCtx2;
+    BossVa* this = THIS;
+    EnBoom* boomerang;
+    s32 i;
+
+    this->actionFunc(this, globalCtx);
+
+    switch (this->actor.params) {
+        case BOSSVA_BODY:
+            if (THIS->colliderBody.base.acFlags & AC_HIT) {
+                this->colliderBody.base.acFlags &= ~AC_HIT;
+                boomerang = (EnBoom*)this->colliderBody.base.ac;
+                if (boomerang->actor.id == ACTOR_EN_BOOM) {
+                    boomerang->returnTimer = 0;
+                }
+            }
+
+            BossVa_UpdateEffects(globalCtx);
+
+            for (i = 2; i >= 0; i--) {
+                if ((globalCtx->envCtx.unk_8C[0][i] - 1) > 0) {
+                    globalCtx->envCtx.unk_8C[0][i] -= 1;
+                } else {
+                    globalCtx->envCtx.unk_8C[0][i] = 0;
+                }
+
+                if ((globalCtx->envCtx.unk_8C[1][i] - 10) > 0) {
+                    globalCtx->envCtx.unk_8C[1][i] -= 10;
+                } else {
+                    globalCtx->envCtx.unk_8C[1][i] = 0;
+                }
+
+                if ((globalCtx->envCtx.unk_8C[2][i] - 10) > 0) {
+                    globalCtx->envCtx.unk_8C[2][i] -= 10;
+                } else {
+                    globalCtx->envCtx.unk_8C[2][i] = 0;
+                }
+            }
+
+            if (this->onCeiling > 0) {
+                this->onCeiling--; // not used by body
+            }
+            break;
+
+        default:
+            this->timer2++;
+            this->actor.focus.pos = this->actor.world.pos;
+            this->actor.focus.pos.y += 45.0f;
+            this->unk_1D8.y = (Math_CosS(this->timer2 * 0xFA4) * 0.24f) + 0.76f;
+            this->unk_1D8.x = (Math_SinS(this->timer2 * 0xFA4) * 0.2f) + 1.0f;
+            break;
+
+        case BOSSVA_SUPPORT_1:
+        case BOSSVA_SUPPORT_2:
+        case BOSSVA_SUPPORT_3:
+        case BOSSVA_ZAPPER_1:
+        case BOSSVA_ZAPPER_2:
+        case BOSSVA_ZAPPER_3:
+        case BOSSVA_DOOR:
+            break;
+    }
+}
+
+s32 BossVa_BodyOverrideLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot,
+                                void* thisx) {
+    BossVa* this = THIS;
+    s32 pad;
+
+    OPEN_DISPS(globalCtx->state.gfxCtx, "../z_boss_va.c", 4156);
+
+    if (limbIndex == 20) {
+        gDPPipeSync(POLY_OPA_DISP++);
+        gSPSegment(POLY_OPA_DISP++, 0x08,
+                   Gfx_TwoTexScroll(globalCtx->state.gfxCtx, 0, 0, 0, 8, 0x10, 1, 0,
+                                    (globalCtx->gameplayFrames * -2) % 64, 0x10, 0x10));
```

Comment:
```suggestion
                   Gfx_TwoTexScroll(globalCtx->state.gfxCtx, 0, 0, 0, 8, 16, 1, 0,
                                    (globalCtx->gameplayFrames * -2) % 64, 16, 16));
```
all `Gfx_TwoTexScroll` args can be decimal (they're positions and sizes)
there are 4 other `Gfx_TwoTexScroll` calls in the file